ast/check.go

```diff
@@ -348,11 +346,11 @@ func (tc *typeChecker) checkExprBuiltin(env *TypeEnv, expr *Expr) *Error {
 func (tc *typeChecker) checkExprEq(env *TypeEnv, expr *Expr) *Error {
 
 	pre := getArgTypes(env, expr.Operands())
-	exp := Equality.Decl.Args()
+	exp := Equality.Decl.FuncArgs()
 
-	if len(pre) < len(exp) {
+	if len(pre) < len(exp.Args) {
 		return newArgError(expr.Location, expr.Operator(), "too few arguments", pre, exp)
-	} else if len(exp) < len(pre) {
+	} else if len(exp.Args) < len(pre) {
```

Comment:
[nit] drop the else? (I'll stop calling these out, it looks like the chance is rather consistent about it, so I'll assume it's intentional.)